src/chocolatey.benchmark/helpers/ManagedProcessHelper.cs

```diff
@@ -0,0 +1,233 @@
+﻿using System;
+using System.Diagnostics;
+using System.Linq;
+using chocolatey.infrastructure.information;
+
+namespace chocolatey.benchmark.helpers
+{
+    internal static class ManagedProcessHelper
+    {
+        private static readonly string[] _filteredParents = new[]
+        {
+            "explorer",
+            "powershell",
+            "pwsh",
+            "cmd",
+            "bash",
+            // The name used to launch windows services
+            // in the operating system.
+            "services",
+            // Known Terminal Emulators
+            "Tabby",
+            "WindowsTerminal",
+            "FireCMD",
+            "ConEmu64",
+            "ConEmuC64"
+        };
+
+        public static ProcessTree GetProcessTree(Process process = null)
+        {
+            if (process == null)
+            {
+                process = Process.GetCurrentProcess();
+            }
+
+            var tree = new ProcessTree(process.ProcessName);
+
+            Process nextProcess = null;
+
+            while (true)
+            {
+                var processId = nextProcess?.Id ?? process.Id;
+                var processName = FindIndexedProcessName(processId);
+
+                if (string.IsNullOrEmpty(processName))
+                {
+                    break;
+                }
+
+                var foundProcess = FindPidFromIndexedProcessName(processName);
+                
+                if (foundProcess == null)
+                {
+                    break;
+                }
+
+                nextProcess = foundProcess;
+                tree.Processes.AddLast(nextProcess.ProcessName);
+            }
+
+            return tree;
+        }
+
+        public static string GetParent(Process process = null)
+        {
+            if (process == null)
+            {
+                process = Process.GetCurrentProcess();
+            }
+
+            Process nextProcess = null;
+
+            while (true)
+            {
+                var processId = nextProcess?.Id ?? process.Id;
+
+                var processName = FindIndexedProcessName(processId);
+
+                if (string.IsNullOrEmpty(processName))
+                {
+                    break;
+                }
+
+                var foundProcess = FindPidFromIndexedProcessName(processName);
+
+                if (foundProcess == null)
+                {
+                    break;
+                }
+
+                nextProcess = foundProcess;
+            }
+
+            return nextProcess?.ProcessName;
+
+            //var processName = FindIndexedProcessName(process.Id);

```

Comment:
Are these comments needed?

Reply:
This was the initial attempt when investigating gathering the process information.

This commented out code would only be useful from a reference point of view, but is not needed anymore